### PR TITLE
Improve session persistence, error handling, and account deletion functionality

### PR DESCRIPTION
**Pull request by MBHunter88**
This pull request introduces enhancements to user session persistence, account deletion functionality, and error handling to improve user experience and ensure reliable session management across page reloads.

Commit Summary:

Implement persistent user session using localStorage for token and user data
- Stores the authentication token and user data in localStorage upon successful login.
- On app load, checks localStorage for an existing token and user data. If found, it initializes the user state to maintain the session.
- Handles session expiration on logout by clearing localStorage and user state.

Ensure newly created users and previous users can delete account based on the id parameter
- Adjusts the DELETE request logic to dynamically use user.id or user.user_id, supporting both newly created and existing users for account deletion.

Enhance error handling in signup modal and move token verification to profile edit block
- Improves the user experience in signup modal by providing specific feedback if a duplicate email error occurs, prompting users to use a different email or log in if they already have an account.
- Refactors token verification to apply only when editing a profile, ensuring that new users don’t encounter a session expiration warning during signup.

**Sessions continue if page is refreshed:**

https://github.com/user-attachments/assets/121840fb-81e8-4138-8462-eecc4f7cb3db

**Duplicate email warning modal:** 
duplicate_error message

### CODE REVIEW

File: client/src/pages/UserProfile.jsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const UserProfile = () => {
             onOk: async () => {
                 try {
                     const token = localStorage.getItem('token');
-                    const response = await fetch(`${import.meta.env.VITE_URL}/users/delete/${user.id}`, {
+                    const response = await fetch(`${import.meta.env.VITE_URL}/users/delete/${user.id || user.user_id}`, {
```
Comment (mai-repo) [inline client/src/pages/UserProfile.jsx:37]: good use of an environment variable

File: client/src/components/SignUpModal.jsx [hunk 2]
```diff
@@ -29,79 +29,84 @@ const SignUpModal = ({ isEditMode = false, initialValues = {}, isSignUpModalOpen
...
-      values.pronouns = customPronoun;
-    }
-    if (isPartnerCustomPronoun) {
-      values.partner_pronouns = partnerCustomPronoun;
-    }
+ // Log user in or update based on input
+ const handleSubmit = async (values) => {
+  // Assign custom pronouns if needed
+  if (isCustomPronoun) {
+    values.pronouns = customPronoun;
```
Comment (mai-repo) [inline client/src/components/SignUpModal.jsx:36]: Also, good naming practices!

File: client/src/components/SignUpModal.jsx [hunk 3]
```diff
@@ -29,79 +29,84 @@ const SignUpModal = ({ isEditMode = false, initialValues = {}, isSignUpModalOpen
...
+        title: 'Sign-Up Failed',
+        content: isDuplicateEmailError ? 'This email address is already in use. Please use a different email or log in if you already have an account.' : errorData.message || 'An unexpected error occurred. Please try again.',
       });
     }
-  };
+  } catch (error) {
+    console.error('Error processing request:', error);
+    Modal.error({
+      title: 'Network Error',
+      content: 'Unable to complete the request. Please try again.',
```
Comment (mai-repo) [inline client/src/components/SignUpModal.jsx:105]: Good use of error handling!

**Review by mai-repo — APPROVED** [review]
Overall, I commend you on creating clean and descriptive code that is easy to read and will be easy! Keep up the good work. :)

**Review by Shikha2410 — APPROVED** [review]

**Pull request merged**